services/device-registry-base/src/main/java/org/eclipse/hono/service/management/credentials/CommonSecret.java

```diff
@@ -160,4 +160,12 @@ public void checkValidity() {
             }
         }
     }
+
+    void merge(final CommonSecret secret) {
```

Comment:
My point was to add an *empty* method at this level which would then be overridden by PasswordSecret and PskSecret. The way you have implemented it couples the generic CommonSecrets to some of its subclasses ...

Reply:
I have added an empty method now and overriden that in the `PasswordSecret` and `PskSecret`.